Source/Data/HAEntity+CompressedEntity.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+
+extension HAEntity {
+    func updatedEntity(compressedEntityState: CompressedEntityState) -> HAEntity? {
+        try? HAEntity(
+            entityId: entityId,
+            domain: domain,
+            state: compressedEntityState.state,
+            lastChanged: compressedEntityState.lastChangedDate ?? lastChanged,
+            lastUpdated: compressedEntityState.lastUpdatedDate ?? lastUpdated,
+            attributes: (compressedEntityState.attributes as? [String: Any]) ?? attributes.dictionary,
+            context: .init(id: compressedEntityState.context ?? "", userId: nil, parentId: nil)
+        )
+    }
+
+    func updatedEntity(adding compressedEntityState: CompressedEntityState) -> HAEntity? {
+        var newAttributes = attributes.dictionary
+        (compressedEntityState.attributes as? [String: Any])?.forEach({ key, value in
+            newAttributes[key] = value
+        })
```

Comment:
the dance around attributes here makes me think you should probably make the HAEntityAttributes know about what you are trying to do, instead